crates/nostr-sqldb/src/model.rs

```diff
@@ -0,0 +1,72 @@
+use diesel::prelude::*;
+use nostr::event::Event;
+use nostr_database::{DatabaseError, FlatBufferBuilder, FlatBufferEncode};
+
+#[cfg(feature = "postgres")]
+use crate::schema::postgres::{event_tags, events};
+
+/// DB representation of [`Event`]
+#[derive(Queryable, Selectable, Insertable, AsChangeset, Debug, Clone)]
+#[diesel(table_name = events)]
+pub struct EventDb {
+    pub id: String,
+    pub pubkey: String,
+    pub created_at: i64,
+    pub kind: i64,
+    pub payload: Vec<u8>,
+    pub signature: String,
+    pub deleted: bool,
+}
+
+/// DB representation of [`EventTag`]
+#[derive(Queryable, Selectable, Insertable, AsChangeset, Debug, Clone)]
+#[diesel(table_name = event_tags)]
+pub struct EventTagDb {
+    pub tag: String,
+    pub tag_value: String,
+    pub event_id: String,
+}
+
+/// A data container for extracting data from [`Event`] and its tags
+#[derive(Debug, Clone)]
+pub struct EventDataDb {
+    pub event: EventDb,
+    pub tags: Vec<EventTagDb>,
+}
+
+impl TryFrom<&Event> for EventDataDb {
+    type Error = DatabaseError;
+    fn try_from(value: &Event) -> Result<Self, Self::Error> {
+        let serialized = value.encode(&mut FlatBufferBuilder::new()).to_vec();
```

Comment:
If the `payload` field will be removed, ignore this comment. Otherwise, the `FlatBufferBuilder` should be constructed somewhere else. It should be constructed once and re-used.

Reply:
Is there some performance gain there with resetting instead of creating a new instance? Do you work with a Mutex there?

Reply:
> Is there some performance gain there with resetting instead of creating a new instance

Yeah, the reset just clear the internal state but keep the capacity, so on the next encoding it doesn't have to re-allocate. The new instance have to allocate every time.

In `nostr-lmdb` I'm currently using an ingester (a separated thread) just for saving events, so I'm not using the Mutex anymore. But yes, in this case the Mutex is ok.